pkg/bgp/mock/gen.go

```diff
@@ -189,8 +190,8 @@ func GenTestEndpointsPairs() (k8s.Endpoints, slim_corev1.Endpoints, metallbspr.E
 	backend := k8s.Backend{
 		NodeName: NodeName,
 	}
-	backends := map[string]*k8s.Backend{
-		IP: &backend,
+	backends := map[cmtypes.AddrCluster]*k8s.Backend{
+		cmtypes.MustParseAddrCluster(IP): &backend,
```

Comment:
There seems to be a lot of additional parsing and addr -> string formatting in loops dealing with backends because of this change. This may have a performance (CPU/memory) impact. Did you consider these options:

1. Keep using a string (with the new @<cluster> allowed) as the key, or
2. Make the new AddrCluster type keep the string format of the address so that re-formatting the same address again and again would be avoided?

Reply:
For option 1, I think it is not a good idea to mix up bare IP string and IP + ClusterID string notation in the code because it is very easy to "leak" that string to the function that only assumes bare IP strings like `net.ParseIP` and get an unintended result. With the `AddrCluster` type, we have a type guarantee. so that we won't make such a mistake.

For option 2, I think it has a performance benefit at this point because we tend to treat IP addresses with `string` and `net.IP` in the majority of places, and we need a conversion between them and `AddrCluster`. However, it is not the case in the future, I think. IIUC, our long-term direction is to convert `net.IP` and `string` into `netip.Addr` (some folks already started to adopt it https://github.com/cilium/cilium/pull/20303, https://github.com/cilium/cilium/compare/master...pr/tklauser/convert-to-netip). So, in the future, the majority of the code becomes to use `netip.Addr` as an internal representation of the IP address. At that point, we shouldn't need many type conversions because there are fewer type gaps.

Of course, the above doesn't answer your concern about the potential performance degradation of "now", but I think even if we change the type to string, it will cause the same degradation at some point. Then, I think it is ok for now to allow it. Maybe I'm too optimistic?